ui/hooks/useAssetDetails.js

```diff
@@ -0,0 +1,118 @@
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getCollectibles, getTokens } from '../ducks/metamask/metamask';
+import { ERC1155, ERC721, ERC20 } from '../helpers/constants/common';
+import {
+  calcTokenAmount,
+  getAssetDetails,
+  getTokenAddressParam,
+  getTokenValueParam,
+} from '../helpers/utils/token-util';
+import { getTransactionData } from '../helpers/utils/transactions.util';
+import { getTokenList } from '../selectors';
+import { hideLoadingIndication, showLoadingIndication } from '../store/actions';
+import { usePrevious } from './usePrevious';
+
+export function useAssetDetails(tokenAddress, userAddress, transactionData) {
+  const dispatch = useDispatch();
+
+  // state selectors
+  const tokens = useSelector(getTokens);
+  const collectibles = useSelector(getCollectibles);
+  const tokenList = useSelector(getTokenList);
+
+  // in-hook state
+  const [currentAsset, setCurrentAsset] = useState(null);
+
+  // previous state checkers
+  const prevTokenAddress = usePrevious(tokenAddress);
+  const prevUserAddress = usePrevious(userAddress);
+  const prevTransactionData = usePrevious(transactionData);
+
+  useEffect(() => {
+    async function getAndSetAssetDetails() {
+      dispatch(showLoadingIndication());
+      const assetDetails = await getAssetDetails(
+        tokenAddress,
+        userAddress,
+        transactionData,
+        collectibles,
+        tokens,
+        tokenList,
+      );
+      setCurrentAsset(assetDetails);
+      dispatch(hideLoadingIndication());
+    }
+    if (
+      tokenAddress !== prevTokenAddress ||
+      userAddress !== prevUserAddress ||
+      transactionData !== prevTransactionData
+    ) {
+      getAndSetAssetDetails();
+    }
```

Comment:
Will using `tokenAddress, userAddress, transactionData` as dependency not suffice above condition ?